### ISSUE TITLE
PTT gets stuck when combined with modifiers

### ISSUE THREAD

**Issue by 9001** [mumble-voip/mumble#1670]
I use CapsLock as my PTT key. As long as CTRL is held down when CapsLock is released, the PTT will not deactivate. For instance, hold CTRL while tapping CapsLock. PTT is enabled and stays active.

Windows 7 x64, Mumble 1.2.8 stable.

**Comment by mkrautz** [comment]
This is probably either because of the nature of the capslock key being a toggle.

Or because of a bad interaction with the toggle when using DirectInput + hooks.

One thing to try before digging into the code would be to disable winhooks in GlobalShortcut_win by adding the following to your registry:

To HKEY_CURRENT_USER\Software\Mumble\Mumble
Add a string value "winhooks", and set its value to "false".

If running with that does work, then it is a bad interaction between the two code paths. If not, it is probably simply because of the nature of the toggle button...

**Comment by mkrautz** [comment]
I also think the title of this issue is wrong.  I'm pretty sure the issue is just capslock, and not a combination of capslock + modifier.  Can you confirm?

**Comment by 9001** [comment]
No, the title is correct - as long as no modifiers are pressed at the same time, then CapsLock acts like any other PTT key. It only happens when it is combined with CTRL, and maybe other modifiers (haven't tested extensively yet). Everything works as expected when the WASD and Space key is used in combination with CapsLock, for instance.

**Comment by mkrautz** [comment]
Interesting!  We've had reports about caps lock before, but never made the connection to modifiers.  Maybe that's the key to the solution. :+1:
